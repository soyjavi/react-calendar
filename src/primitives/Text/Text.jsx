import PropTypes from 'prop-types';
import React from 'react';

import { styles } from '../helpers';
import style from './Text.module.css';

const Text = ({ action, bold, children, headline, small, tag = 'span', upperCase, ...inherit }) =>
  React.createElement(
    tag,
    {
      className: styles(
        style.primitive,
        bold && style.bold,
        headline ? style.headline : action ? style.action : small ? style.small : style.paragraph,
        upperCase && style.upperCase,
        inherit.className,
      ),
    },
    children,
  );

Text.displayName = 'Primitive:Text';

Text.propTypes = {
  action: PropTypes.bool,
  bold: PropTypes.bool,
  children: PropTypes.string,
  headline: PropTypes.bool,
  upperCase: PropTypes.bool,
  small: PropTypes.bool,
  tag: PropTypes.string,
};

export { Text };
