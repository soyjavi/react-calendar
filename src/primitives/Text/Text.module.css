.primitive {
  color: var(--content);
  font-family: var(--font);
  font-weight: var(--font-weight);
  text-align: var(--text-align);
}

.bold {
  font-family: var(--font-bold);
  font-weight: var(--font-bold-weight);
}

.headline {
  font-size: var(--font-size-headline);
}

.paragraph {
  font-size: var(--font-size-paragraph);
}

.action {
  font-size: var(--font-size-action);
}

.small {
  font-size: var(--font-size-small);
}

.upperCase {
  text-transform: uppercase;
}
