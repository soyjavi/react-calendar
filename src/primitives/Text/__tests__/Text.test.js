import { render } from '@soyjavi/testing-react';
import React from 'react';

import { Text } from '../Text';

const DEFAULTS = {
  children: 'Lorem Ipsum',
};

describe('primitive:<Text>', () => {
  test('renders', () => {
    const { toJSON } = render(<Text {...DEFAULTS} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('prop:action', () => {
    const { toJSON } = render(<Text {...DEFAULTS} action />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('prop:bold', () => {
    const { toJSON } = render(<Text {...DEFAULTS} bold />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('prop:headline', () => {
    const { toJSON } = render(<Text {...DEFAULTS} headline />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('prop:upperCase', () => {
    const { toJSON } = render(<Text {...DEFAULTS} upperCase />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('prop:small', () => {
    const { toJSON } = render(<Text {...DEFAULTS} small />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('prop:h1', () => {
    const { toJSON } = render(<Text {...DEFAULTS} tag="h1" />);
    expect(toJSON()).toMatchSnapshot();
  });

  // Testing
  test('testID', () => {
    const { toJSON } = render(<Text {...DEFAULTS} data-testID="reactor" />);
    // expect(getByTestId('reactor')).toBeDefined();
    expect(toJSON()).toMatchSnapshot();
  });

  // Styling
  test('inherit:className', () => {
    const { toJSON } = render(<Text {...DEFAULTS} className="reactor" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
