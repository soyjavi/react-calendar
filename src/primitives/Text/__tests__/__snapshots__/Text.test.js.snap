// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`primitive:<Text> inherit:className 1`] = `
<DocumentFragment>
  <span
    class="primitive paragraph reactor"
  >
    Lorem Ipsum
  </span>
</DocumentFragment>
`;

exports[`primitive:<Text> prop:action 1`] = `
<DocumentFragment>
  <span
    class="primitive action"
  >
    Lorem Ipsum
  </span>
</DocumentFragment>
`;

exports[`primitive:<Text> prop:bold 1`] = `
<DocumentFragment>
  <span
    class="primitive bold paragraph"
  >
    Lorem Ipsum
  </span>
</DocumentFragment>
`;

exports[`primitive:<Text> prop:h1 1`] = `
<DocumentFragment>
  <h1
    class="primitive paragraph"
  >
    Lorem Ipsum
  </h1>
</DocumentFragment>
`;

exports[`primitive:<Text> prop:headline 1`] = `
<DocumentFragment>
  <span
    class="primitive headline"
  >
    Lorem Ipsum
  </span>
</DocumentFragment>
`;

exports[`primitive:<Text> prop:small 1`] = `
<DocumentFragment>
  <span
    class="primitive small"
  >
    Lorem Ipsum
  </span>
</DocumentFragment>
`;

exports[`primitive:<Text> prop:upperCase 1`] = `
<DocumentFragment>
  <span
    class="primitive paragraph upperCase"
  >
    Lorem Ipsum
  </span>
</DocumentFragment>
`;

exports[`primitive:<Text> renders 1`] = `
<DocumentFragment>
  <span
    class="primitive paragraph"
  >
    Lorem Ipsum
  </span>
</DocumentFragment>
`;

exports[`primitive:<Text> testID 1`] = `
<DocumentFragment>
  <span
    class="primitive paragraph"
  >
    Lorem Ipsum
  </span>
</DocumentFragment>
`;
